deb/brave-keyring/DEBIAN/postinst

```diff
@@ -1,4 +1,14 @@
-#!/bin/sh
+#!/bin/sh -e
 
-# Delete (expired) key added by old install instructions
-apt-key --keyring /etc/apt/trusted.gpg del D8BAD4DE7EE17AF52A834B2D0BB75829C2D4E821
+[ "$1" = configure ] || exit
+
+# Match if the new keyring in /usr/share/keyrings is in use.
+new_keyring_pattern='^[[:space:]]*deb[^#]*signed-by=/usr/share/keyrings/brave-browser-archive-keyring.gpg[^#]*brave-browser-apt-release.s3.brave.com'
+
+# Delete the old keyring file if the new one is already in use.
+# Otherwise replace the old file with a symlink to its new location.
+if grep -q "${new_keyring_pattern:?}" /etc/apt/sources.list.d/brave-browser-release.list; then
+    rm -f /etc/apt/trusted.gpg.d/brave-browser-release.gpg
```

Comment:
If this matches and this file does not exist, the script will throw an error here (but exit 0 because `-f`). I know it works, but I can imagine a user filing a bug report about this error being thrown on a brand new installation.
```
# apt install ./brave-keyring_1.11-1.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'brave-keyring' instead of './brave-keyring_1.11-1.deb'
The following NEW packages will be installed:
  brave-keyring
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/6500 B of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 /deb/output/brave-keyring_1.11-1.deb brave-keyring all 1.11 [6500 B]
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package brave-keyring.
(Reading database ... 4786 files and directories currently installed.)
Preparing to unpack .../brave-keyring_1.11-1.deb ...
Unpacking brave-keyring (1.11) ...
Setting up brave-keyring (1.11) ...
grep: /etc/apt/sources.list.d/brave-browser-release.list: No such file or directory
```

Reply:
Added a test for the existence of that file.